javaparser-testing/src/test/java/com/github/javaparser/ast/validator/Java7ValidatorTest.java

```diff
@@ -39,6 +39,18 @@ public void tryWithoutAnything() {
     }
 
     @Test
+    public void tryWithResourceVariableDeclaration() {
+        ParseResult<Statement> result = javaParser.parse(STATEMENT, provider("try(Reader r = new Reader()){}"));
+        assertNoProblems(result);
+    }
+
+    @Test
+    public void tryWithResourceReference() {
+        ParseResult<Statement> result = javaParser.parse(STATEMENT, provider("try(a.b.c){}"));
+        assertProblems(result, "(line 1,col 1) Try with resources only supports variable declarations.");
```

Comment:
@ftomassetti check this out - much better than "PARSE ERROR AT 1,1" right?

Reply:
Nice!